runtime/src/main/java/com/redhat/ipaas/rest/ComponentGroups.java

```diff
@@ -37,12 +38,15 @@
     @Inject
     private DataManager dataMgr;
 
+    @Context
+    private UriInfo uri;
+
     @GET
     @Produces(MediaType.APPLICATION_JSON)
     @ApiOperation(value = "List component groups")
     @ApiResponses(value = {@ApiResponse(code = 200, message = "Success", response = ComponentGroup.class)})
     public Collection<ComponentGroup> list() {
-        return dataMgr.fetchAll(ComponentGroup.class);
+        return dataMgr.fetchAll(ComponentGroup.class, new ReflectiveSorter<>(ComponentGroup.class, uri.getQueryParameters()));
```

Comment:
Yeah I'd rather separate that out. How about an intermediate `SortOptions` to pass to the `ReflectiveSorter` that's created from the context?